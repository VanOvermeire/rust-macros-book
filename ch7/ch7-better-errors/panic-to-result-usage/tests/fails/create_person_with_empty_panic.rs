use panic_to_result_macro::panic_to_result;

#[derive(Debug)]
pub struct Person {
    name: String,
    age: u32,
}

#[panic_to_result]
fn create_person_with_empty_panic(name: String, age: u32) -> Person {
    if age > 30 {
        panic!();
    }
    Person {
        name,
        age,
    }
}

fn main() {}
