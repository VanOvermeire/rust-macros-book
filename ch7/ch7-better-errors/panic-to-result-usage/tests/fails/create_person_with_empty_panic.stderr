error: please make sure every panic in your function has a message
  --> tests/fails/create_person_with_empty_panic.rs:12:9
   |
12 |         panic!();
   |         ^^^^^
